test/Microsoft.TestPlatform.SmokeTests/DataCollectorTests.Coverlet.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.TestPlatform.SmokeTests
+{
+    using Microsoft.TestPlatform.TestUtilities;
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using System;
+    using System.IO;
+    using System.Linq;
+    using System.Reflection;
+
+    [TestClass]
+    public class DataCollectorTestsCoverlets : IntegrationTestBase
+    {
+        [TestMethod]
+        public void RunCoverletCoverage()
+        {
+            string projectFileName = this.GetProjectAssetFullPath("CoverletCoverageTestProject.csproj", "CoverletCoverageTestProject.csproj");
+            string logId = Guid.NewGuid().ToString("N");
+            this.InvokeDotnetTest($"--collect:\"XPlat Code Coverage\" {projectFileName} --diag:coverletcoverage.{logId}.log");
+            var currentDir = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location);
+
+            // Verify vstest.console.dll CollectArgumentProcessor fix codeBase for coverlet package
```

Comment:
Not sure what this comment means, could you rephrase it to be more concrete, or explain to me what it means in a comment here?

Reply:
With this PR https://github.com/microsoft/vstest/pull/2288 we removed custom code to allow coverlet lib to load in-proc collector.
That PR added some code inside `CollectArgumentProcessor` where we looking for "base path" of `coverlet.collector.dll` expected inside nuget package default folder.
When we find the lib we'll emit a log https://github.com/microsoft/vstest/pull/2288/files#diff-6f2506b5205b023825192043bd33a256R138
This assert check that we're sure that we've updated collector setting code base with full path, otherwise without "custom coverlet code" inside `ProxyExecutionManager` coverlet dll won't be resolved inside testhost.